src/dtos/view-dto.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable import/no-cycle */
+import { Error } from '../models/error';
+
+import { DatasetDTO } from './dataset-dto';
+
+export interface PageInfo {
+    total_records: number | undefined;
+    start_record: number | undefined;
+    end_record: number | undefined;
+}
+
+export interface ViewErrDTO {
+    success: boolean;
+    errors: Error[];
+    dataset_id: string;
+}
+
+export interface ViewDTO {
+    success: boolean;
+    dataset: DatasetDTO;
+    current_page: number;
+    page_info: PageInfo;
+    pages: Array<string | number>;
+    page_size: number;
+    total_pages: number;
+    headers: Array<string> | undefined;
+    data: Array<Array<string>>;
+}
```

Comment:
So we don't yet have any visualiser or processing logic yet...  This "View" returns the current page of the dataset along with useful information about the dataset to the front end so it can be rendered.  We're still working from the fact we currently only have a single CSV to return to the user.  We load it to memory and then create a 3D array of the data that we return to the user.  Currently, even the headers are read directly off the CSV file.  I hope to move this code to using the headers from the database in later revisions.